pype/plugins/standalonepublisher/publish/collect_shots.py

```diff
@@ -0,0 +1,125 @@
+import os
+from urllib.parse import unquote, urlparse
+
+import opentimelineio as otio
+from bson import json_util
+
+import pyblish.api
+from pype import lib
+from avalon import io
+
+
+class OTIO_View(pyblish.api.Action):
+    """Currently disabled because OTIO requires PySide2. Issue on Qt.py:
+    https://github.com/PixarAnimationStudios/OpenTimelineIO/issues/289
+    """
+
+    label = "OTIO View"
+    icon = "wrench"
+    on = "failed"
+
+    def process(self, context, plugin):
+        instance = context[0]
+        representation = instance.data["representations"][0]
+        file_path = os.path.join(
+            representation["stagingDir"], representation["files"]
+        )
+        lib._subprocess(["otioview", file_path])
+
+
+class CollectShots(pyblish.api.InstancePlugin):
+    """Collect Anatomy object into Context"""
+
+    order = pyblish.api.CollectorOrder
+    label = "Collect Shots"
+    hosts = ["standalonepublisher"]
+    families = ["editorial"]
+    actions = []
+
+    def process(self, instance):
+        representation = instance.data["representations"][0]
+        file_path = os.path.join(
+            representation["stagingDir"], representation["files"]
+        )
+        timeline = otio.adapters.read_from_file(file_path)
+        tracks = timeline.each_child(
+            descended_from_type=otio.schema.track.Track
+        )
+        asset_entity = instance.context.data["assetEntity"]
+        asset_name = asset_entity["name"]
+
+        # Project specific prefix naming. This needs to be replaced with some
+        # options to be more flexible.
+        asset_name = asset_name.split("_")[0]
```

Comment:
I'm not sure how this could possible be flexible for the other productions. Current client has parent asset something like `[project_code]101_shots`, but the shots need to labelled `[project_code]101_[shot_name]`

Reply:
yeah this is a tricky one. That might be solved once we make proper GUI for editorial ingesting outside of standalone publisher.